### PR TITLE
fixed relevel bug

### PR DESCRIPTION
**Pull request by WackerO**
This PR finally (!) makes relevel functional

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/rnadeseq/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/rnadeseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: assets/RNAseq_report.Rmd [hunk 1]
```diff
@@ -553,6 +550,9 @@ if (isProvided(params$path_relevel)) {
     }
 }
 
+# Run actual DESeq analysis
+cds <- DESeq(cds)
```
Comment (SusiJo) [inline assets/RNAseq_report.Rmd:554]: What about `parallel = FALSE` ? as in line 452

File: tests/test_relevel.yml [hunk 2]
```diff
@@ -3,12 +3,12 @@
   tags:
     - test_relevel
   files:
-    - path: results_test/differential_gene_expression/DE_genes_tables/DE_contrast_condition_genotype_WT_vs_KO.tsv
-      md5sum: 93fd663cb976c102144e1ffb34d88e53
+    - path: results_test/differential_gene_expression/DE_genes_tables/DE_contrast_condition_genotype_KO_vs_WT.tsv
```
Comment (SusiJo) [inline tests/test_relevel.yml:6]: Why is it now `KO_vs_WT` while it ws the other way round before?

File: tests/test_relevel.yml [hunk 3]
```diff
@@ -43,15 +43,15 @@
     - path: results_test/differential_gene_expression/plots/further_diagnostics_plots/Effects_of_transformations_on_the_variance.pdf
     - path: results_test/differential_gene_expression/plots/further_diagnostics_plots/Effects_of_transformations_on_the_variance.png
     - path: results_test/differential_gene_expression/plots/further_diagnostics_plots/Effects_of_transformations_on_the_variance.svg
-    - path: results_test/differential_gene_expression/plots/further_diagnostics_plots/histogram_of_p.valuescondition_genotype_WT_vs_KO.pdf
-    - path: results_test/differential_gene_expression/plots/further_diagnostics_plots/histogram_of_p.valuescondition_genotype_WT_vs_KO.png
-    - path: results_test/differential_gene_expression/plots/further_diagnostics_plots/histogram_of_p.valuescondition_genotype_WT_vs_KO.svg
+    - path: results_test/differential_gene_expression/plots/further_diagnostics_plots/histogram_of_p.valuescondition_genotype_KO_vs_WT.pdf
```
Comment (SusiJo) [inline tests/test_relevel.yml:46]: would be nice to have an underscore here `values_condition_...` like below for `rejections_...`

File: assets/RNAseq_report.Rmd [hunk 4]
```diff
@@ -537,13 +535,12 @@ if (params$input_type == "featurecounts") {
...
         rownames(coldata) <- qbicCodes
         cds <- DESeqDataSetFromTximport(txi=txi.salmon, colData =coldata, design = eval(parse(text=as.character(design[[1]]))))
-        cds <- DESeq(cds)
     }
 } else {
     stop(paste0("Invalid input type: ", params$input_type, "! Input type must be one of: featurecounts, rsem, salmon!"))
 }
 
-#Apply relevel if provided to metadata
+#Apply relevel if provided; this must be done before calling DESeq()
```
Comment (SusiJo) [inline assets/RNAseq_report.Rmd:543]: :D yes! 💯

File: tests/test_relevel.yml (hunk 2 above)
Comment (WackerO) [reply tests/test_relevel.yml:6]: So the relevel works now; before, it didn't. I have to say however that I noticed the order of factors is utterly confusing. I specifically looked this up; by default, orders of factors are alphabetically; after releveling, the reference should to my understanding be the first factor. This is not the case, but it does change compared to the normal test profile, I tested that

File: tests/test_relevel.yml (hunk 3 above)
Comment (WackerO) [reply tests/test_relevel.yml:46]: done

File: tests/test_relevel.yml (hunk 2 above)
Comment (SusiJo) [reply tests/test_relevel.yml:6]: Hmm I think that after releveling it should be `condition_of_interest_VS_condition_reference`, so the new naming should actually be correct.

**Comment by SusiJo** [comment]
At some point you should tackle some of the linting problems.

**Comment by WackerO** [comment]
Of course, I'll deal with that in the template update :)
Thanks for reviewing!

**Pull request merged**